### PR TITLE
feat: enhance destroy subcommands & new workflow test destroy

### LINKED ISSUE #3345: epic: Fix destroy commands
## Description

Currently destroy commands are unreliable and are not able to handle removal of partially deployed components, once they run into any kind of issue during execution they directly fail, causing a deadlock in a lot of scenarios 

- [ ] Relay
- [ ] Consensus node
- [ ] Explorer
- [ ] Mirror node
- [ ] Block node

## Depends on: https://github.com/hiero-ledger/solo/issues/3352

### PR DESCRIPTION
**Pull request by JeffreyDallas**
## Description

This pull request changes the following:

* Updated individual components destroy command and One shot destroy command, to be more robust and can move on if one step is failing. 
* Add a new test workflow to test one shot deployment being interrupted at different random stages. It uses a timeout command to kill the one-shot deployment command at a random time. And then try destroy everything and redeploy everything. 

### Related Issues

* Closes #3345
* Related to #

### Pull request (PR) checklist

* \[ ] This PR added tests (unit, integration, and/or end-to-end)
* \[ ] This PR updated documentation
* \[ ] This PR added no TODOs or commented out code
* \[ ] This PR has no breaking changes
* \[ ] Any technical debt has been documented as a separate issue and linked to this PR
* \[ ] Any `package.json` changes have been explained to and approved by a repository manager
* \[ ] All related issues have been linked to this PR
* \[ ] All changes in this PR are included in the description
* \[ ] When this PR merges the commits will be squashed and the title will be used as the commit message, the 'commit message guidelines' below have been followed

### Testing

* \[ ] This PR added unit tests
* \[ ] This PR added integration/end-to-end tests
* \[ ] These changes required manual testing that is documented below
* \[ ] Anything not tested is documented

The following manual testing was done:

* TBD

The following was not tested:

* TBD

<details>
<summary>
Commit message guidelines
</summary>
We use 'Conventional Commits' to ensure that our commit messages are easy to read, follow a consistent format, and for automated release note generation. Please follow the guidelines below when writing your commit messages:

1. BREAKING CHANGE: a commit that has a footer BREAKING CHANGE:, or appends a ! after the type/scope, introduces a breaking API change (correlating with MAJOR in Semantic Versioning). A BREAKING CHANGE can be part of commits of any type.  NOTE: currently breaking changes will only bump the MAJOR version.
2. The title is prefixed with one of the following:

| Prefix    | Description                                         | Semantic Version Update | Captured in Release Notes |
|-----------|-----------------------------------------------------|-------------------------|---------------------------|
| feat:     | a new feature                                       | MINOR                   | Yes                       |
| fix:      | a bug fix                                           | PATCH                   | Yes                       |
| perf:     | performance                                         | PATCH                   | Yes                       |
| refactor: | code change that isn't feature or fix               | none                    | No                        |
| test:     | adding missing tests                                | none                    | No                        |
| docs:     | changes to documentation                            | none                    | Yes                        |
| build:    | changes to build process                            | none                    | No                        |
| ci:       | changes to CI configuration                         | none                    | No                        |
| style:    | formatting, missing semi-colons, etc                | none                    | No                        |
| chore:    | updating grunt tasks etc; no production code change | none                    | No                        |

</details>

### CODE REVIEW

File: .github/workflows/flow-one-shot-interrupt-destroy.yaml [hunk 1]
```diff
@@ -0,0 +1,139 @@
...
+        if: always()
+        timeout-minutes: 15
+        env:
+          ONE_SHOT_WITH_BLOCK_NODE: true
+        run: |
+          if [ "${{ inputs.released-or-latest || 'false' }}" = "true" ]; then
+            npx @hashgraph/solo one-shot single destroy --quiet-mode || true
+          else
+            npm run solo -- one-shot single destroy --quiet-mode || true
+          fi
```
Comment (jeromy-cannon) [inline .github/workflows/flow-one-shot-interrupt-destroy.yaml]:
```suggestion
          if [ "${{ inputs.released-or-latest || 'false' }}" = "true" ]; then
            npx @hashgraph/solo one-shot single destroy --quiet-mode
          else
            npm run solo -- one-shot single destroy --quiet-mode
          fi
```
destroy should not fail.  probably needs my changes in: https://github.com/hiero-ledger/solo/pull/3427

File: examples/address-book/README.md [hunk 2]
```diff
@@ -8,6 +8,9 @@ NOTE: Mirror Node refers to File 102 as its address book.
 
 ### Download Archive
 
+> **Note for unzipped release archive:** If you run tasks from an unzipped release directory (outside the Solo git repository), set `USE_RELEASED_VERSION` before running `task`:
+> `export USE_RELEASED_VERSION=true`
+
```
Comment (jeromy-cannon) [inline examples/address-book/README.md]: No.  The default should be to be able to run from any directory.  We should enhance our CI to doing a local build and set env variables as needed.

User experience must come first over our Solo developer experience.

File: examples/external-database-test/README.md [hunk 3]
```diff
@@ -15,6 +15,9 @@ This example demonstrates how to deploy a Hiero Hashgraph Solo network with an e
 
 ### Download Archive
 
+> **Note for unzipped release archive:** If you run tasks from an unzipped release directory (outside the Solo git repository), set `USE_RELEASED_VERSION` before running `task`:
+> `export USE_RELEASED_VERSION=true`
+
```
Comment (jeromy-cannon) [inline examples/external-database-test/README.md]: No.  The default should be to be able to run from any directory.  We should enhance our CI to doing a local build and set env variables as needed.

User experience must come first over our Solo developer experience.

File: examples/hardhat-with-solo/README.md [hunk 4]
```diff
@@ -12,6 +12,9 @@ This example demonstrates how to deploy a Hiero Hashgraph Solo deployment via th
 
 ### Download Archive
 
+> **Note for unzipped release archive:** If you run tasks from an unzipped release directory (outside the Solo git repository), set `USE_RELEASED_VERSION` before running `task`:
+> `export USE_RELEASED_VERSION=true`
+
```
Comment (jeromy-cannon) [inline examples/hardhat-with-solo/README.md]: No.  The default should be to be able to run from any directory.  We should enhance our CI to doing a local build and set env variables as needed.

User experience must come first over our Solo developer experience.

File: examples/running-solo-inside-cluster/README.md [hunk 5]
```diff
@@ -1,5 +1,8 @@
 # Running Solo Inside Cluster Example
 
+> **Note for unzipped release archive:** If you run tasks from an unzipped release directory (outside the Solo git repository), set `USE_RELEASED_VERSION` before running `task`:
+> `export USE_RELEASED_VERSION=true`
+
```
Comment (jeromy-cannon) [inline examples/running-solo-inside-cluster/README.md]: No.  The default should be to be able to run from any directory.  We should enhance our CI to doing a local build and set env variables as needed.

User experience must come first over our Solo developer experience.

File: src/commands/mirror-node.ts [hunk 6]
```diff
@@ -1363,12 +1366,14 @@ VALUES (decode('${exchangeRates}', 'hex'), ${timestamp + '000001'}, ${exchangeRa
...
-              this.configManager.getFlag<DeploymentName>(flags.deployment),
-              this.configManager.getFlag<boolean>(flags.forcePortForward),
-            );
+            if (remoteConfigLoaded) {
+              await this.accountManager.loadNodeClient(
+                context_.config.namespace,
+                this.remoteConfig.getClusterRefs(),
+                this.configManager.getFlag<DeploymentName>(flags.deployment),
+                this.configManager.getFlag<boolean>(flags.forcePortForward),
+              );
```
Comment (jeromy-cannon) [inline src/commands/mirror-node.ts]: In my PR I killed this.  I could not figure out a reason why mirror node destroy would need to load the node client, which is time consuming due to the port forward logic.  it also fails if the consensus nodes are not functioning correctly.

File: examples/network-with-domain-names/README.md [hunk 7]
```diff
@@ -12,6 +12,9 @@ This example demonstrates how to deploy a Hiero Hashgraph Solo network with cust
 
 ### Download Archive
 
+> **Note for unzipped release archive:** If you run tasks from an unzipped release directory (outside the Solo git repository), set `USE_RELEASED_VERSION` before running `task`:
+> `export USE_RELEASED_VERSION=true`
+
```
[the remark this answers is not in the document]
Comment (JeffreyDallas) [reply examples/network-with-domain-names/README.md]: removed, will address in new PR to enhance CI

File: src/commands/one-shot/default-one-shot.ts [hunk 8]
```diff
@@ -1073,6 +1124,7 @@ export class DefaultOneShotCommand extends BaseCommand implements OneShotCommand
           // set up the sub-tasks
           return task.newListr(subTasks, {
             concurrent: true,
+            exitOnError: false,
```
Comment (jeromy-cannon) [inline src/commands/one-shot/default-one-shot.ts]: Keep running all tasks even if there is a failure?
